Automattic/newspack-newsletters
# Use custom capability_type for the CPT

### All Submissions:






### Changes proposed in this Pull Request:



Registers the newsletter CPT with a custom `capability_type`, in order to allow granular control over access to this CPT.

Also adds a small tweak for handling "Other" provider in some methods. 

### How to test the changes in this Pull Request:

1. As an administrator or editor user, ensure that the following can all be edited:
    2. Newsletters
    3. Newsletter Ads
    4. Newsletter Ads Advertisers
    5. Subscription Lists (via Newspack's Engagement wizard UI)
    6. Layouts (via Newsletters post editor)  
2. Install the [capability-manager-enhanced plugin](https://wordpress.org/plugins/capability-manager-enhanced/) and create a new role in "Roles" submenu item
3. Go to "Capabilities" submenu item and select the new role. Select "Newspack Newsletters" group and assign all the caps

<img width="1048" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/7383192/8286e17d-4d5d-4792-b592-16a5197c8edd">

5. Due to a quirk in CME plugin, also assign these same caps in the "Editing" group:

<img width="441" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/7383192/3de8b9ed-1abb-434d-bcbd-b1ad60fb2673">

6. Select "Newspack" group and assign the engagement wizard cap:

<img width="706" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/7383192/3c940042-45a1-404f-b5d3-8485b6c60068">

8. Assign `edit_posts` (in Posts group) and `read` in Admin group caps -> this way the user will be able to log in to /wp-admin
9. Save the caps
10. Create a new user, with the newly created role, and log in as them – then redo first step of these instructions 

### Other information:
